BE: Auth: Add OAuth2 proxy support for authentication requests

joshuaNathaniel opened the pull request:
**What changes did you make?** (Give an overview)

This PR adds OAuth2 proxy support by configuring the OAuth2 WebClient to respect JVM system proxy properties, #196 . The implementation is minimal and uses Reactor Netty's built-in `proxyWithSystemProperties()` method.

Key changes:
- Modified `OAuthSecurityConfig` to configure OAuth2 WebClient with system proxy support
- Uses standard JVM proxy properties (`-Dhttps.proxyHost`, `-Dhttps.proxyPort`, etc.) 
- No custom configuration required - works automatically when system properties are set
- Fully backward compatible - no proxy used when properties aren't set 

**Is there anything you'd like reviewers to focus on?**

The implementation is intentionally minimal - just 4 lines of actual code that leverage Reactor Netty's built-in proxy support. 

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)
<!-- ignore-task-list-start -->
- [x] Integration
- [x] Manually
<!-- ignore-task-list-end -->

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/kafbat/kafka-ui/blob/main/.github/CONTRIBUTING.md) and [Code of Conduct](https://github.com/kafbat/kafka-ui/blob/main/.github/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**
🦦


Closes #196

joshuaNathaniel committed:
feat: Add OAuth2 proxy support for authentication requests

Fixes #196

- Add SimpleOAuthProxyConfig to handle proxy configuration for OAuth2
- Support both system proxy properties and explicit proxy configuration
- Wire proxy-aware WebClient into OAuth2 user services
- Add unit tests following repository patterns

OAuth2 authentication now respects proxy settings when enabled, allowing
Kafbat UI to work behind corporate firewalls. The feature is disabled by
default to maintain backward compatibility.

Haarolean commented on api/src/main/java/io/kafbat/ui/config/auth/SimpleOAuthProxyConfig.java:
```diff
@@ -0,0 +1,53 @@
+  }
+
+  @Bean(name = "oauth2WebClient")
+  @ConditionalOnProperty(value = "auth.oauth2.proxy.enabled", havingValue = "true")
```
why do we need to support a proxy via custom settings rather than using a default system one provided by jre? Another point, in case we'd need a proxy somewhere else, we'd need to create yet another configuration with another bunch of properties to support proxy in multiple places.

joshuaNathaniel committed:
feat: Add OAuth2 proxy support using JVM system properties

Simplify OAuth2 authentication to automatically respect JVM proxy settings
(e.g., -Dhttps.proxyHost, -Dhttps.proxyPort). The implementation uses Reactor
Netty's built-in proxyWithSystemProperties() method, which gracefully handles
both proxy and non-proxy scenarios without additional configuration.

joshuaNathaniel committed:
feat: Extend OAuth2 proxy support to all authentication endpoints

Add proxy-aware WebClient configuration for JWT decoder, opaque token
introspection, and OIDC authentication manager. All OAuth2 HTTP calls
now respect JVM system proxy properties.

- Add ReactiveJwtDecoder bean with proxy-aware JWKS fetching
- Add ReactiveOpaqueTokenIntrospector bean with proxy support
- Add ReactiveOAuth2AccessTokenResponseClient for token endpoint
- Configure OidcAuthorizationCodeReactiveAuthenticationManager
- Add WireMock integration tests for with/without proxy scenarios

joshuaNathaniel committed:
refactor: Make OAuth2 WebClient a qualified Spring bean

joshuaNathaniel committed:
test: fix issue with netty proxy not allowing localhost as proxy on linux

joshuaNathaniel committed:
test: provide additional comments and guards for oauth test support

joshuaNathaniel commented on api/src/test/java/io/kafbat/ui/config/auth/OAuthTestSupport.java:
```diff
@@ -0,0 +1,279 @@
+      System.setProperty("http.proxyHost", "localhost");
+      System.setProperty("http.proxyPort", String.valueOf(proxyServer.port()));
+      // Override default nonProxyHosts ("localhost|127.*|[::1]") to allow proxying localhost for tests
+      System.setProperty("http.nonProxyHosts", "none");
```
We test locally. Add HTTPS is overhead for very little R.O.I.

joshuaNathaniel commented on api/src/main/java/io/kafbat/ui/config/auth/OAuthSecurityConfig.java:
```diff
@@ -84,8 +107,47 @@ public SecurityWebFilterChain configure(ServerHttpSecurity http, OAuthLogoutSucc
+    String jwkSetUri = getJwkSetUri();
+    if (jwkSetUri == null) {
+      return token -> Mono.error(new IllegalStateException("JWT decoder not configured"));
+    }
```
This has been tested without issue. Adding additional constraints provides no value.

joshuaNathaniel replied:
The separate jwtDecoder bean was originally created to inject the proxy-enabled WebClient as the inline jwkSetUri() method doesn't accept a custom WebClient. However, I've reverted to inline config using jwtDecoder() instead of jwkSetUri(), which allows passing the WebClient directly while keeping the original code structure. This minimizes the diff while still enabling proxy support.

joshuaNathaniel commented on api/src/main/java/io/kafbat/ui/config/auth/OAuthSecurityConfig.java:
```diff
@@ -84,8 +107,47 @@ public SecurityWebFilterChain configure(ServerHttpSecurity http, OAuthLogoutSucc
+        webClient.mutate()
+            .defaultHeaders(h -> h.setBasicAuth(config.getClientId(), config.getClientSecret()))
+            .build());
+  }
```
This has been tested without issue. Again, no additional value.

joshuaNathaniel committed:
refactor: Use inline OAuth2 resource server config with proxy WebClient

Address review feedback by reverting to inline configuration style
for JWT decoder and opaque token introspector. The proxy-enabled
WebClient is now injected directly into configure() and used inline,
removing the need for separate bean methods.

- Remove jwtDecoder and opaqueTokenIntrospector beans
- Remove getJwkSetUri and getOpaqueTokenConfig helper methods
- Remove redundant JwtDecoder/OpaqueToken test classes

germanosin committed:
Merge branch 'main' into issues/196

Haarolean reviewed (approved):
Thank you very much for your contribution!

Haarolean committed:
Merge branch 'main' into issues/196

krisurnok commented:
Hi,
We are using a proxy application that authenticates users via OAuth and sets the access token in a cookie.

We need guidance on how to configure this setup. Below are the details we are using:

Proxy application URL: http://localhost:3000

Issuer URL: different from the proxy URL (for example, Auth0: https://domain.eu.auth0.com/)

JWKS URI: https://domain.eu.auth0.com/.well-known/jwks.json

Could you please advise on the correct configuration for this scenario?
